numba/targets/arraymath.py

```diff
@@ -345,14 +286,261 @@ def flat_all(a):
 @overload_method(types.Array, "any")
 def np_any(a):
     def flat_any(a):
-        for v in a.flat:
-            if v:
+        for v in numpy.nditer(a):
+            if v.item():
                 return True
         return False
 
     return flat_any
 
 
+def get_isnan(dtype):
+    """
+    A generic isnan() function
+    """
+    if isinstance(dtype, types.Number):
+        return numpy.isnan
+    else:
+        @jit(nopython=True)
+        def isnan(x):
+            return False
+
+        return isnan
+
+
+@overload(numpy.nanmin)
+def np_nanmin(a):
+    if not isinstance(a, types.Array):
+        return
+    isnan = get_isnan(a.dtype)
+
+    def nanmin_impl(a):
+        if a.size == 0:
+            raise ValueError("nanmin(): empty array")
+        for view in numpy.nditer(a):
+            minval = view.item()
+            break
+        for view in numpy.nditer(a):
+            v = view.item()
+            if not minval < v and not isnan(v):
```

Comment:
for comparison operations with nan, they ought return false so the branch ought to execute anyway without the second guard condition?